.github/workflows/build.yml

```diff
@@ -98,3 +98,63 @@ jobs:
       - name: Setup tmate debug session
         uses: mxschmitt/action-tmate@v3
         if: env.RUN_TMATE
+  test:
+    name: test source - go${{ matrix.go-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+        go-version:
+          - "1.19"
+          - "1.20"
+        include:
+          - os: ubuntu-20.04
+            go-version: "1.20"
+    steps:
+      - uses: actions/checkout@v3
+      - id: setup-go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go-version }}
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-golang-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-golang-
+
+      - name: Retrieve package location
+        run: |
+            echo PACKAGE_LOCATION="$(find ./ -name 'test')/../" >> $GITHUB_OUTPUT
+        id: package-location
+      - name: Install go-junit-report
+        env:
+          PACKAGE_URL: github.com/jstemmer/go-junit-report
+          PACKAGE_VERSION: ${{ steps.setup-env.outputs.go-junit-version }}
+        run: go install ${PACKAGE_URL}@${PACKAGE_VERSION}
+      - name: Install go-mock
+        env:
+          PACKAGE_URL: github.com/golang/mock/mockgen
+          PACKAGE_VERSION: ${{ steps.setup-env.outputs.go-mock-version }}
+        run: go install ${PACKAGE_URL}@${PACKAGE_VERSION}
+
+      - name: Generate mocks and run tests
+        working-directory: ${{ steps.package-location.outputs.PACKAGE_LOCATION }}
+        run: |
+          export GOPATH="$HOME/go/"
+          export PATH=$PATH:$GOPATH/bin
+          go generate ./
+          go test -v ./test/* | go-junit-report -set-exit-code > report.xml
+
```

Comment:
Anybody have any thoughts about including [staticcheck](https://staticcheck.io/docs/running-staticcheck/ci/github-actions/) in our test workflow?

Reply:
It is a hook option for [TekWizely/pre-commit-golang](https://github.com/TekWizely/pre-commit-golang) so I would rather bake this into the linting configuration.

Reply:
I'm for it.

Reply:
Please see PR https://staticcheck.io/docs/running-staticcheck/ci/github-actions/ which implements this

Reply:
Excellent, thanks!